core/src/com/mygdx/battleship/BattleshipBots/SmartRandomBot.java

```diff
@@ -0,0 +1,220 @@
+                for (int x = 1; x < shipSizes[curShip]; x++) {
+                    int newR = startR + x;
+                    String newPlace = String.valueOf(Character.toUpperCase(startL)) + newR;
+                    if (usedCoordinates.contains(newPlace) || !BattleshipUtils.validateCoordinate(newPlace)){
+                        reset = true;
+                        break;
+                    } else {
+                        placement[x] = newPlace;
+                    }
+                }
+            } else if (orientation == 2) { // south
+                for (int x = 1; x < shipSizes[curShip]; x++) {
+                    char newL = (char) (startL + x);
+                    String newPlace = String.valueOf(Character.toUpperCase(newL)) + startR;
+                    if (usedCoordinates.contains(newPlace) || !BattleshipUtils.validateCoordinate(newPlace)){
+                        reset = true;
+                        break;
+                    } else {
+                        placement[x] = newPlace;
+                    }
+                }
+            } else { // west
+                for (int x = 1; x < shipSizes[curShip]; x++) {
+                    int newR = startR - x;
+                    String newPlace = String.valueOf(Character.toUpperCase(startL)) + newR;
+                    if (usedCoordinates.contains(newPlace) || !BattleshipUtils.validateCoordinate(newPlace)){
+                        reset = true;
+                        break;
+                    } else {
+                        placement[x] = newPlace;
+                    }
+                }
+            }
+
+            //check invariants
+            reset = reset || !BattleshipUtils.validateShip(placement);
+            if (!reset) {
+                shipsLocations[curShip] = placement;
+                curShip++;
+                usedCoordinates.addAll(Arrays.asList(placement));
+            }
+        }
+    }
+
+	/**
+	 * Randomly decides next move.
+     * If hit in last turn, randomly choose a direction and continue on direction
+     * If another hit, continue on direction.
+     * If it has found a ship, will intelligently shoot around the ship until it dies.
+	 */
+	private void nextMove() {
+        if ((sink)||(!foundShip)) {
+            move = getRandomMove();
+        } else {
+            //if miss, select an untried direction, otherwise keep direction.
+            if (!hit) {
+                usedDirections.add(direction);
+                int newDirection = getOpposite(direction); //prioritize opposite direction
+                Random rand = new Random();
+                while (usedDirections.contains(newDirection)) {
+                    newDirection = rand.nextInt();
+                }
+                direction = newDirection;
+            }
+            char left = move.charAt(0);
+            int right = Integer.parseInt(move.substring(1, move.length()));
+            if (direction == 0) { // north
```

Comment:
extract 0,1,2,3 as either static final int constants named NORTH, EAST, SOUTH, WEST, or use an enum that accomplishes the same thing, just for code cleanliness